src/nacl/public.py

```diff
@@ -92,17 +92,18 @@ def __init__(self, private_key, encoder=encoding.RawEncoder):
 
         self._private_key = private_key
         self.public_key = PublicKey(raw_public_key)
+        self._hashbytes = b'p:' + bytes(self.public_key)
 
     def __bytes__(self):
         return self._private_key
 
     def __hash__(self):
-        return hash(bytes(self))
+        return hash(self._hashbytes)
```

Comment:
I've done so for using the same "seed" for __eq__ and __hash__ while keeping sure a PrivateKey and its corresponding PublicKey could not by any means end in the same hashing bucket if they were used as keys in a dict.